cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -358,11 +351,13 @@ public void getAppInfo(HttpRequest request, HttpResponder responder,
   @Path("/apps/{app-id}/versions")
   public void listAppVersions(HttpRequest request, HttpResponder responder,
                               @PathParam("namespace-id") final String namespaceId,
-                              @PathParam("app-id") final String appId) throws Exception {
-    ApplicationId applicationId = validateApplicationId(namespaceId, appId);
-    Collection<String> versions = applicationLifecycleService.getAppVersions(namespaceId, appId);
+                              @PathParam("app-id") final String appName) throws Exception {
+    // The version of the validated applicationId is ignored. We only use the method to validate the input.
+    validateApplicationId(namespaceId, appName);
+    Collection<String> versions = applicationLifecycleService.getAppVersions(new NamespaceId(namespaceId),
+                                                                             appName);
     if (versions.isEmpty()) {
-      throw new ApplicationNotFoundException(applicationId);
+      throw new ApplicationNotFoundException(new ApplicationId(namespaceId, appName));
```

Comment:
Please create a separate constructor for the exception that takes only namespace and appName. current one would include "-SNAPSHOT" in the message and this would be confusing